.github/workflows/build.yml

```diff
@@ -0,0 +1,51 @@
+name: Build
+
+on: [ push, pull_request, workflow_dispatch ]
+
+jobs:
+  java:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: "Node Cache"
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-cache-node-${{ hashFiles('**/pom.xml') }}
+
+      - name: "Node Modules Cache"
+        uses: actions/cache@v2
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-cache-node_modules-${{ hashFiles('**/pom.xml') }}
+
+      - name: "Maven Cache"
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-cache-maven-${{ hashFiles('**/pom.xml') }}
```

Comment:
This is not what I meant. I don't think this is correct either. We still want to lookup exact cache if package.json has not changed. But if package.json or pom.xml change it should fall back to a restore key such as:

```
         key: ${{ runner.os }}-cache-maven-${{ hashFiles('**/pom.xml') }}
          restore-keys: ${{ runner.os }}-cache-maven-
```

and

```
          key: ${{ runner.os }}-cache-node_modules-${{ hashFiles('**/package.json') }}
          restore-keys: ${{ runner.os }}-cache-node_modules-
```